tests/unit/basic.tcl

```diff
@@ -291,89 +292,105 @@ start_server {tags {"basic"}} {
-#        r rename mykey1 mykey2
-#        r get mykey2
-#    } {hello}
+    test {RENAME against already existing key} {
+        r set mykey a
+        r set mykey2 b
+        r rename mykey2 mykey
+        set res [r get mykey]
+        append res [r exists mykey2]
+    } {b0}
 
-#    test {RENAME source key should no longer exist} {
-#        r exists mykey
-#    } {0}
+    test {RENAMENX basic usage} {
+        r del mykey
+        r del mykey2
+        r set mykey foobar
+        r renamenx mykey mykey2
+        set res [r get mykey2]
+        append res [r exists mykey]
+    } {foobar0}
+
+     test {RENAMENX against already existing key} {
+        r set mykey foo
+        r set mykey2 bar
+        r renamenx mykey mykey2
+    } {0}
 
-#    test {RENAME against already existing key} {
-#        r set mykey a
-#        r set mykey2 b
-#        r rename mykey2 mykey
-#        set res [r get mykey]
-#        append res [r exists mykey2]
-#    } {b0}
+    test {RENAMENX against already existing key (2)} {
+        set res [r get mykey]
+        append res [r get mykey2]
+    } {foobar}
 
-#    test {RENAMENX basic usage} {
-#        r del mykey
-#        r del mykey2
-#         r set mykey foobar
-#         r renamenx mykey mykey2
-#         set res [r get mykey2]
-#         append res [r exists mykey]
-#     } {foobar0}
-#
-#     test {RENAMENX against already existing key} {
-#         r set mykey foo
-#         r set mykey2 bar
-#         r renamenx mykey mykey2
-#     } {0}
-#
-#     test {RENAMENX against already existing key (2)} {
-#        set res [r get mykey]
-#        append res [r get mykey2]
-#    } {foobar}
-#
-#    test {RENAME against non existing source key} {
-#        catch {r rename nokey foobar} err
-#        format $err
-#    } {ERR*}
-#
-#    test {RENAME where source and dest key is the same} {
-#        catch {r rename mykey mykey} err
-#        format $err
-#    } {ERR*}
-#
-#    test {RENAME with volatile key, should move the TTL as well} {
-#        r del mykey mykey2
-#        r set mykey foo
-#        r expire mykey 100
-#        assert {[r ttl mykey] > 95 && [r ttl mykey] <= 100}
-#        r rename mykey mykey2
-#        assert {[r ttl mykey2] > 95 && [r ttl mykey2] <= 100}
-#    }
-#
-#    test {RENAME with volatile key, should not inherit TTL of target key} {
-#        r del mykey mykey2
-#        r set mykey foo
-#        r set mykey2 bar
-#        r expire mykey2 100
-#        assert {[r ttl mykey] == -1 && [r ttl mykey2] > 0}
-#        r rename mykey mykey2
-#        r ttl mykey2
-#    } {-1}
+    test {RENAME against non existing source key} {
+        catch {r rename nokey foobar} err
+        format $err
+    } {ERR*}
+
+    test {RENAMENX against non existing source key} {
+        catch {r renamenx nokey foobar} err
+        format $err
+    } {ERR*}
+
+    test {RENAME where source and dest key are the same (existing)} {
+        r set mykey foo
+        r rename mykey mykey
+    } {OK}
```

Comment:
In fact, RENAME where source and dest key are the same (existing) is allowed. https://github.com/redis/redis/blame/f95031c4733078788063de775c968b6dc85522c0/tests/unit/keyspace.tcl#L134